source/extensions/transport_sockets/tls/context_config_impl.cc

```diff
@@ -327,27 +363,22 @@ ServerContextConfigImpl::ServerContextConfigImpl(
                         DEFAULT_CIPHER_SUITES, DEFAULT_CURVES, factory_context),
       require_client_certificate_(
           PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, require_client_certificate, false)),
-      session_ticket_keys_([&config, &api = api_] {
-        std::vector<SessionTicketKey> ret;
-
-        switch (config.session_ticket_keys_type_case()) {
-        case envoy::api::v2::auth::DownstreamTlsContext::kSessionTicketKeys:
-          for (const auto& datasource : config.session_ticket_keys().keys()) {
-            validateAndAppendKey(ret, Config::DataSource::read(datasource, false, api));
-          }
-          break;
-        case envoy::api::v2::auth::DownstreamTlsContext::kSessionTicketKeysSdsSecretConfig:
-          throw EnvoyException("SDS not supported yet");
-          break;
-        case envoy::api::v2::auth::DownstreamTlsContext::SESSION_TICKET_KEYS_TYPE_NOT_SET:
-          break;
-        default:
-          throw EnvoyException(fmt::format("Unexpected case for oneof session_ticket_keys: {}",
-                                           config.session_ticket_keys_type_case()));
-        }
-
-        return ret;
-      }()) {
+      session_ticket_keys_provider_(getTlsSessionTicketKeysConfigProvider(
+          factory_context, config, &session_ticket_keys_dynamic_sds_)) {
+  if (session_ticket_keys_provider_ != nullptr && session_ticket_keys_dynamic_sds_) {
```

Comment:
Seems session_ticket_keys_dynamic_sds_ only used here? Can you just pass the validation callback to `getTlsSessionTicketKeysConfigProvider` so it can be added when it is needed?